tests/kafkatest/services/performance/jmx_mixin.py

```diff
@@ -0,0 +1,80 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+class JmxMixin(object):
+
+    def __init__(self, num_nodes, jmx_object_name=None, jmx_attributes=None):
+        self.jmx_object_name = jmx_object_name
+        self.jmx_attributes = jmx_attributes
+        self.jmx_port = 9192
+
+        self.started = [False] * num_nodes
+        self.jmx_stats = [{} for x in range(num_nodes)]
+        self.maximum_jmx_value = {}  # map from object_attribute_name to maximum value observed over time
+        self.average_jmx_value = {}  # map from object_attribute_name to average value observed over time
+
+    def clean_node(self, node):
+        node.account.kill_process("jmx", clean_shutdown=False, allow_fail=True)
+        node.account.ssh("rm -rf /mnt/jmx_tool.log", allow_fail=False)
+
+    def maybe_start_jmx_tool(self, idx, node):
+        if self.started[idx-1] == True or self.jmx_object_name == None:
+            return
+        self.started[idx-1] = True
+
+        cmd = "/opt/kafka/bin/kafka-run-class.sh kafka.tools.JmxTool " \
+              "--reporting-interval 1000 --jmx-url service:jmx:rmi:///jndi/rmi://127.0.0.1:%d/jmxrmi" % self.jmx_port
+        if self.jmx_object_name != None:
+            cmd += " --object-name %s" % self.jmx_object_name
+        if self.jmx_attributes != None:
+            cmd += " --attributes %s" % self.jmx_attributes
+        cmd += " > /mnt/jmx_tool.log &"
+
+        self.logger.debug("Start JmxTool %d command: %s", idx, cmd)
+        node.account.ssh(cmd, allow_fail=False)
```

Comment:
We've found it's very helpful with debugging test failures to provide some sort of fail-fast guarantee that the process we care about is up and running before returning from these start commands

E.g.
```
def start_jmx_tool(...):
   node.account.ssh(cmd...) 
   wait_until(jmx_tool_is_alive)
   # now it's ok to return
```